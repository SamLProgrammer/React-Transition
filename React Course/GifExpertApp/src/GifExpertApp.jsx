import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
    const [categories, setCategories] = useState(['Dragon Ball']);

    const handleAddCategory = (category) => {
        if (categories.find((cat) => { return cat.toLowerCase() == category.toLowerCase() })) return;
        setCategories((cat) => { return [category, ...categories] });
    }

    return (
        <>
            <h1>GifExpertApp</h1>
            <AddCategory onAddCategory={handleAddCategory} />
            {
                categories.map((category) => <GifGrid category={category} limit={5} key={category}/>)
            }
        </>
    )
}
