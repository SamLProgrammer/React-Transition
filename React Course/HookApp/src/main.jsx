import React from 'react'
import ReactDOM from 'react-dom/client'
import './index.css'
import { MainApp } from './useContext/MainApp'
import './useReducer/intro-reducer'
import { BrowserRouter, createBrowserRouter, RouterProvider, } from "react-router-dom";

ReactDOM.createRoot(document.getElementById('root')).render(
    //   // <React.StrictMode>
    <BrowserRouter>
        <MainApp />
    </BrowserRouter>
    //   //  </React.StrictMode>,
)
