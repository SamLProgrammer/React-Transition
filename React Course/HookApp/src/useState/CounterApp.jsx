import { useState } from "react"

export const CounterApp = () => {
  const [counter, setCounter] = useState({
    counter1: 10,
    counter2: 20,
    counter3: 30,
  });

  const handleAddClick = () => {
    setCounter((counter) => ({ ...counter, counter1: counter.counter1 + 1 }));
  }
  return (
    <>
      <h1>Counter</h1>
      <h2>{counter.counter1}</h2>
      <h2>{counter.counter2}</h2>
      <h2>{counter.counter3}</h2>
      <hr></hr>
      <button onClick={handleAddClick}>+1</button>
    </>
  )
}
