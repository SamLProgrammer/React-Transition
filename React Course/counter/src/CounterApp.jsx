import PropTypes from 'prop-types'
import { useState } from 'react';

export const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState(value);
    
    const handleAddClick = (evt) => {
        setCounter((c) => c+1);
    }

    const handleSubClick = (evt) => {
        setCounter((c) => c-1);
    }

    const handleResetClick = (evt) => {
        setCounter(value);
    }

    return (
        <>
            <h1>Counter App</h1>
            <h2>{counter}</h2>
            <button onClick={(evt) => handleAddClick(evt)}>+1</button>
            <button onClick={(evt) => handleSubClick(evt)}>-1</button>
            <button onClick={(evt) => handleResetClick(evt)}>Reset</button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}