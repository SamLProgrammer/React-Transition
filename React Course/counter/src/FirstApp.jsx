import PropTypes from 'prop-types'

export const FirstApp = ({ title, subTitle }) => {
    return (
        <>
            <h1>First App</h1>
            <h2>Hey {title}</h2>
            <h3>You are: {subTitle}</h3>
        </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.number
}