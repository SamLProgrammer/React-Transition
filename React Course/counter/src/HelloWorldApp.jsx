export const App = () => {
    return (
        <h1>Hola ke ase!!</h1>
    );
}

