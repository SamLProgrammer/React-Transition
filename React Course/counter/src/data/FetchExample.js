export const fetchingCatFact = async () => {
    
    const fact = await fetch('https://catfact.ninja/fact');
    return await fact.json();
}