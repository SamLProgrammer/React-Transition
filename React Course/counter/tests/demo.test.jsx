import { render, screen, fireEvent } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";
import { fetchingCatFact } from "../src/data/FetchExample";
import { FirstApp } from "../src/FirstApp";

describe('First Test', () => { 
    test('Cat Facts', async() => { 
        render(<CounterApp value={0}/>);
        fireEvent.click(screen.getByText('+1'));
        expect(screen.getByRole('heading', {level: 2}).innerHTML).toBe('1');
     })
 })