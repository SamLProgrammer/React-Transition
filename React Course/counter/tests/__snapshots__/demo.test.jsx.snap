// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First Test Cat Facts 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    0
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button>
    Reset
  </button>
</div>
`;
